src/library.js

```diff
@@ -6264,16 +6536,25 @@ LibraryManager.library = {
   // netinet/in.h
   // ==========================================================================
 
+#if USE_PTHREADS
+  in6addr_any: '; if (ENVIRONMENT_IS_PTHREAD) _in6addr_any = PthreadWorkerInit._in6addr_any; else PthreadWorkerInit._in6addr_any = _in6addr_any = allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], "i8", ALLOC_STATIC)',
+  in6addr_loopback: '; if (ENVIRONMENT_IS_PTHREAD) _in6addr_loopback = PthreadWorkerInit._in6addr_loopback; else PthreadWorkerInit._in6addr_loopback = _in6addr_loopback = allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1], "i8", ALLOC_STATIC)',
+#else
   in6addr_any:
     'allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], "i8", ALLOC_STATIC)',
   in6addr_loopback:
     'allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1], "i8", ALLOC_STATIC)',
+#endif
 
   // ==========================================================================
   // netdb.h
   // ==========================================================================
 
+#if USE_PTHREADS
```

Comment:
All these lines in src/library.js are more of a temporary thing that I want to work on removing in favor of a properly multithreaded C library. There is a noticeable perf impact, and a chance of deadlocking even(!) with this proxying scheme, so I'm not sure if this would warrant any formalization, but it's best to just get rid of it. Also it would be a bit complex, since there would be nested calls to {{{ }}}, so would have to implement nesting level detection, e.g.

    tzname: '; if (ENVIRONMENT_IS_PTHREAD) _tzname = PthreadWorkerInit._tzname; else PthreadWorkerInit._tzname = _tzname = allocate({{{ 2*Runtime.QUANTUM_SIZE }}}, "i32*", ALLOC_STATIC)',